openpathsampling/tools.py

```diff
@@ -34,18 +34,36 @@ def in_ipynb():
 except ImportError:
     is_ipynb = False
 
+last_output = None
 
 def refresh_output(output_str, print_anyway=True, refresh=True,
                    output_stream=None, ipynb_display_only=False):
+
+    global last_output
+
     if output_stream is None:
         output_stream = sys.stdout
 
-    if is_ipynb or not ipynb_display_only or print_anyway:
-        if refresh:
+    if not output_str:
+        return
+
+    if refresh:
+        if is_ipynb:
             IPython.display.clear_output(wait=True)
+        else:
+            if last_output is not None:
+                lines = len(last_output.split('\n'))
+                CURSOR_UP_ONE = '\x1b[1A'
+                ERASE_LINE = '\x1b[2K'
+
+                output_stream.write((CURSOR_UP_ONE + ERASE_LINE) * (lines - 1))
```

Comment:
Ah yes, I forgot about this. I used it to keep the output short and mimic the behaviour of overwriting the output in a cell. If you write to a file it will contain the control characters and if you run `cat ...` it will _reply_ the output. In an editor you just have the control characters.

I like this, but we might add an additional optional flag.